linear_trainer.py

```diff
@@ -28,12 +32,17 @@ def linear_test(config, model, datasets):
         metrics.update(preds, target)
 
         if k > 0:
-            top_k_idx[slice] = argsort_top_k(preds, k, axis=1)
-            top_k_scores[slice] = np.take_along_axis(
-                preds, top_k_idx[slice], axis=1)
-
+            idx[slice] = argsort_top_k(preds, k, axis=1)
+            value[slice] = np.take_along_axis(preds, idx[slice], axis=1)
+
+        if config.save_positive_predictions:
+            for i_pred in preds:
+                indice = i_pred > 0
```

Comment:
we have
single: index
plural: indexes or indices
let's use pos_idx; see other comments later